op-conductor/rpc/ws/flashblocks_handler.go

```diff
@@ -0,0 +1,325 @@
+	cfg             Config
+	log             log.Logger
+	isLeaderFn      func(context.Context) bool
+	rollupBoostConn *websocket.Conn
+	server          *http.Server
+	hub             *Hub
+}
+
+// NewHandler creates a new flashblocks handler
+func NewHandler(cfg Config, log log.Logger, isLeaderFn func(context.Context) bool) FlashblockHandler {
+	return &Handler{
+		cfg:        cfg,
+		log:        log,
+		isLeaderFn: isLeaderFn,
+		hub:        newHub(),
+	}
+}
+
+// Start initializes the flashblocks handler and starts the WebSocket server and rollup boost listener
+func (h *Handler) Start(ctx context.Context) error {
+	go h.hub.run()
+
+	// Start the WebSocket server
+	if err := h.startWebSocketServer(ctx); err != nil {
+		return err
+	}
+
+	// Start the rollup boost listener
+	if err := h.startRollupBoostListener(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Stop closes any open WebSocket connections and shuts down the server
+func (h *Handler) Stop() {
+	// Signal the hub to stop
+	close(h.hub.done)
+
+	// Close the rollup boost connection if it exists
+	if h.rollupBoostConn != nil {
+		h.rollupBoostConn.Close()
+		h.rollupBoostConn = nil
+	}
+
+	// Force close the HTTP server if it's running
+	if h.server != nil {
+		h.log.Info("closing WebSocket server")
+		err := h.server.Close()
+		if err != nil {
+			h.log.Error("Error closing WebSocket server", "err", err)
+		}
+		h.log.Info("WebSocket server closed")
+	}
+}
+
+// BroadcastMessage sends a message to all connected WebSocket clients
+func (h *Handler) BroadcastMessage(message []byte) {
+	clientCount := len(h.hub.clients)
+	if clientCount == 0 {
+		h.log.Debug("no WebSocket clients connected, not broadcasting message")
+		return
+	}
+
+	h.log.Trace("Broadcasting message to WebSocket clients", "clientCount", clientCount)
+	h.hub.broadcast <- message
+}
+
+// startRollupBoostListener initializes and starts a WebSocket client to listen for rollup boost messages
+func (h *Handler) startRollupBoostListener(ctx context.Context) error {
+	if h.cfg.RollupBoostWsURL == "" {
+		h.log.Info("rollup boost WebSocket disabled, no URL configured")
+		return nil
+	}
+
+	// Start a goroutine to maintain the rollup boost WebSocket connection
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				if h.rollupBoostConn != nil {
+					h.log.Info("closing rollup boost WebSocket connection due to context cancellation")
+					h.rollupBoostConn.Close()
+					h.rollupBoostConn = nil
+				}
+				return
+			case <-h.hub.done:
+				return
+			default:
+				// Try to connect if not connected
+				if h.rollupBoostConn == nil {
+					conn, _, err := websocket.DefaultDialer.Dial(h.cfg.RollupBoostWsURL, nil)
+					if err != nil {
+						h.log.Error("failed to connect to rollup boost WebSocket", "err", err)
+						time.Sleep(time.Second) // delay before retry
+						continue
+					}
+					h.rollupBoostConn = conn
+					h.log.Info("connected to rollup boost WebSocket", "url", h.cfg.RollupBoostWsURL)
+				}
```

Comment:
This should be in the NewHandler function.

We need to make sure it succeeds before return. And only start listening to message when Start is called.